syntax="proto3";

package protocol;

message CSChatMessage
{
  int32 player_id = 1;
  string chat_message = 2;
}

message SCChatMessage
{
  int32 player_id = 1;
  string chat_message = 3;
}

message CSCreateRoom
{
  string room_name = 1;
}

message SCCreateRoom
{
  string room_name = 1;
  int32 room_id = 2;
}

message CSInvitePlayerIntoRoom
{
  int32 room_id = 1;
  int32 player_id = 2;
}

message CSRequestIntoRoom
{
  int32 room_id = 1;
}

message CSResponseInviteIntoRoom
{
  int32 room_id = 1;
  int32 agree = 2;
}

message CSResponseRequestIntoRoom
{
  int32 room_id = 1;
  int32 player_id = 2;
  int32 agree = 3;
}

message SCInviteIntoRoomList
{
  repeated int32 room_id = 1;
}

message SCRequestIntoRoomItem
{
  int32 room_id = 1;
  int32 player_id = 2;
}

message SCRequestIntoRoomList
{
  repeated SCRequestIntoRoomItem request = 1;
}

message CSRoomChatMessage
{
  int32 room_id = 1;
  string chat_message = 2;
}

message SCRoomChatMessage
{
  int32 player_id = 1;
  string player_name = 2;

  int32 room_id = 3;
  string chat_message = 4;
}
